pydra/engine/tests/test_workflow.py

```diff
@@ -1403,52 +1454,198 @@ def test_wf_3nd_ndst_6(plugin, tmpdir):
     assert wf.output_dir.exists()
 
 
-# workflows with structures A -> B -> C with multiple connections
+def test_wf_3nd_ndst_7(plugin, tmpdir):
+    """workflow with three tasks, third one connected to two previous tasks,
+    the third one uses scalar splitter from the previous ones
+    """
+    wf = Workflow(name="wf_ndst_9", input_spec=["x"])
+    wf.add(add2(name="add2x", x=wf.lzin.x).split("x"))
+    wf.add(add2(name="add2y", x=wf.lzin.x).split("x"))
+    wf.add(
+        multiply(name="mult", x=wf.add2x.lzout.out, y=wf.add2y.lzout.out).split(
+            ("_add2x", "_add2y")
+        )
+    )
+    wf.inputs.x = [1, 2]
+    wf.set_output([("out", wf.mult.lzout.out)])
+    wf.cache_dir = tmpdir
 
+    with Submitter(plugin=plugin) as sub:
+        sub(wf)
 
-def test_wf_3nd_7(tmpdir):
-    """workflow with three tasks A->B->C vs two tasks A->C with multiple connections"""
-    wf = Workflow(name="wf", input_spec=["zip"], cache_dir=tmpdir)
-    wf.inputs.zip = [["test1", "test3", "test5"], ["test2", "test4", "test6"]]
+    results = wf.result()
+    assert results.output.out == [9, 16]
+    # checking the output directory
+    assert wf.output_dir.exists()
+
+
+def test_wf_3nd_ndst_8(plugin, tmpdir):
+    """workflow with three tasks, third one connected to two previous tasks,
+    using default splitter in the third task
+    """
+    wf = Workflow(name="wf_ndst_9", input_spec=["x"])
+    wf.add(add2(name="add2x", x=wf.lzin.x).split("x"))
+    wf.add(add2(name="add2y", x=wf.lzin.x).split("x"))
+    wf.add(multiply(name="mult", x=wf.add2x.lzout.out, y=wf.add2y.lzout.out))
+    wf.inputs.x = [1, 2]
+    wf.set_output([("out", wf.mult.lzout.out)])
+    wf.cache_dir = tmpdir
 
-    wf.add(identity_2flds(name="iden2flds_1", x1=wf.lzin.zip, x2="Hoi").split("x1"))
+    with Submitter(plugin=plugin) as sub:
+        sub(wf)
+
+    assert wf.mult.state.splitter == ["_add2x", "_add2y"]
+    assert wf.mult.state.splitter_rpn == ["add2x.x", "add2y.x", "*"]
+    results = wf.result()
+    assert results.output.out == [9, 12, 12, 16]
+    # checking the output directory
+    assert wf.output_dir.exists()
+
+
+# workflows with structures A -> B -> C with multiple connections
 
-    wf.add(identity(name="identity", x=wf.iden2flds_1.lzout.out1))
 
+def test_wf_3nd_multcon_1(plugin, tmpdir):
+    """workflow with three tasks, first one has a splitter
+    second uses the splitter from the first one,
+    the third one uses outer splitter (so the first splitter is "repeated")
+    """
+    wf = Workflow(name="wf_ndst_9", input_spec=["x"])
+    wf.add(add2(name="add2x", x=wf.lzin.x).split("x"))
+    wf.add(add2(name="add2y", x=wf.add2x.lzout.out))
     wf.add(
-        identity_2flds(
-            name="iden2flds_2", x1=wf.identity.lzout.out, x2=wf.iden2flds_1.lzout.out2
+        multiply(name="mult", x=wf.add2x.lzout.out, y=wf.add2y.lzout.out).split(
+            ["_add2x", "_add2y"]
```

Comment:
is this explicit split required and why? how would the user/workflow creator know to do this?

why is this different from: `test_wf_3nd_ndst_8`

Reply:
i see the next test uses default

Reply:
yes, was checking if gives the same when default